apis/python/src/tiledbsoma/_tdb_handles.py

```diff
@@ -220,9 +220,24 @@ def _opener(
         context: SOMATileDBContext,
         timestamp: int,
     ) -> tiledb.Group:
-        return tiledb.Group(
-            uri, mode, ctx=_group_timestamp_ctx(context.tiledb_ctx, timestamp)
-        )
+        # * We want to do open-group-at-timestamp.
+        # * Timestamps are read from tiledb config.
+        # * We can't set-config on an already opened group.
+        # * We could create a new ctx, with the current ctx's config plus the
+        #   new timestamp config, but that causes a thread-pool leak. See also
+        #   https://github.com/single-cell-data/TileDB-SOMA/issues/1169.
+        # * For now, we do this close, set-config, and re-open workaround.
+        # * An upcoming TileDB-Py release will remove this need for us.
+        # TODO: make that happen.
+        # Tracking issue: https://github.com/single-cell-data/TileDB-SOMA/issues/1198
+        G = tiledb.Group(uri, mode)
```

Comment:
I suggest we wait for the next TileDB-Py version where we just take the config on start, and avoid having to open and close.

Reply:
FYI TileDB-Py 0.21.2 was released this morning and includes the addition of a `Group` constructor that takes in a `Config` argument.

Reply:
@nguyenv on it 😎